erpnext/education/doctype/quiz/quiz.py

```diff
@@ -30,25 +31,29 @@ def evaluate(self, response_dict, quiz_name):
 		questions = [frappe.get_doc('Question', question.question_link) for question in self.question]
 		answers = {q.name:q.get_answer() for q in questions}
 		result = {}
-		for key in answers:
-			try:
-				if isinstance(response_dict[key], list):
-					is_correct = compare_list_elementwise(response_dict[key], answers[key])
-				else:
-					is_correct = (response_dict[key] == answers[key])
-			except Exception as e:
-				is_correct = False
-			result[key] = is_correct
-		score = (sum(result.values()) * 100 ) / len(answers)
+
+		# if assessment questions exists then evaluate on the basis of number of questions given in quiz
+		if self.assessment_questions:
+			for question in response_dict.items():
+				result = evaluate_response(answers, response_dict)
```

Comment:
- evaluate_response returns a dict, I think we will have to do dict.update instead of assigning a value directly